Hakawai/Mentions/HKWMentionsPlugin.m

```diff
@@ -1677,6 +1677,15 @@ - (void)textViewDidChangeSelectionV2:(UITextView *)textView {
     if (range.length > 0) {
         return;
     }
+
+    // When cursor is placed within mention, highlight that mention.
+    const NSUInteger cursorLocation = range.location;
+    [self.mentions enumerateObjectsUsingBlock:^(HKWMentionsAttribute *mention, NSUInteger idx __unused, BOOL *stop __unused) {
+        const NSUInteger mentionStartLocation = mention.range.location;
+        const BOOL isCursorInsideMention = (cursorLocation > mentionStartLocation) && (cursorLocation < mentionStartLocation + mention.range.length);
+        [self toggleMentionsFormattingAtRange:mention.range selected:isCursorInsideMention];
```

Comment:
ok so after looking at this, maybe using currentlySelectedMentionRange is good.

So in this function, you can do the following

NSRange range;
[attribute:HKWMentionsAttribute atIndex:range.location effectiveRange:&range];

if range != self.currentlySelectionMentionRange
   [self toggleMentionsFormattingAtRange:currentlySelectionMentionRange selected:NO];
   [self toggleMentionsFormattingAtRange:range selected:YES]; 
   
self.currentlySelectionMentionRange = range

with appropriate guarding

Reply:
Thanks @bkoatz  for reviewing PR.
 
We will be using `attribute(at:effectiveRange:)` as described in https://github.com/linkedin/Hakawai/pull/170#discussion_r461786851

- On selecting mention, we want to highlight that mention and de-select previous selected mention.
- `currentlySelectedMentionRange` holds current mention range only and it's location always start with next location (i.e `+1` ) than actual mention location.
- So, we have to maintain previous selected mention to de-select it or de-select all mentions expect one that is going to be selected.
- So we think that usage of `currentlySelectedMentionRange`  won't be suitable in this case.

Please let us know if we are missing something.

Reply:
I think just whatever range you get from attribute(at:...) you can save for future reference so you can unselect it later. you can save it in a variable currentlySelectedMentionRangeV2, which I referenced in slack.

Reply:
We have tried to implement `attribute(at:effectiveRange:)` but this method has issue with normal text.
`attribute(at:effectiveRange:)` return mention range for normal text as well.
So we have used similar method (i.e `enumerateAttributesInRange(:options:usingBlock)` which also iterates over attributes) and not considers normal text as mention text.

Also introduced `currentlySelectedMentionRangeV2` property.